File: source/clibCoreCs/Models/Entity/0/contentFieldHelpModel.vb

Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses
Imports Contensive.Core.Controllers
Imports Newtonsoft.Json

Namespace Contensive.Core.Models.Entity
    Public Class ContentFieldHelpModel
        Inherits baseModel
        '
        '====================================================================================================
        '-- const
        Public Const contentName As String = "content field help"
        Public Const contentTableName As String = "ccFieldHelp"
        Private Shadows Const contentDataSource As String = "default"
        '
        '====================================================================================================
        ' -- instance properties
        Public Property FieldID As Integer
        Public Property HelpCustom As String
        Public Property HelpDefault As String
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass) As ContentFieldHelpModel
            Return add(Of ContentFieldHelpModel)(cpCore)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function add(cpCore As coreClass, ByRef callersCacheNameList As List(Of String)) As ContentFieldHelpModel
            Return add(Of ContentFieldHelpModel)(cpCore, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer) As ContentFieldHelpModel
            Return create(Of ContentFieldHelpModel)(cpCore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordId As Integer, ByRef callersCacheNameList As List(Of String)) As ContentFieldHelpModel
            Return create(Of ContentFieldHelpModel)(cpCore, recordId, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordGuid As String) As ContentFieldHelpModel
            Return create(Of ContentFieldHelpModel)(cpCore, recordGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function create(cpCore As coreClass, recordGuid As String, ByRef callersCacheNameList As List(Of String)) As ContentFieldHelpModel
            Return create(Of ContentFieldHelpModel)(cpCore, recordGuid, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String) As ContentFieldHelpModel
            Return createByName(Of ContentFieldHelpModel)(cpCore, recordName)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createByName(cpCore As coreClass, recordName As String, ByRef callersCacheNameList As List(Of String)) As ContentFieldHelpModel
            Return createByName(Of ContentFieldHelpModel)(cpCore, recordName, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub save(cpCore As coreClass)
            MyBase.save(cpCore)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, recordId As Integer)
            delete(Of ContentFieldHelpModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Sub delete(cpCore As coreClass, ccGuid As String)
            delete(Of ContentFieldHelpModel)(cpCore, ccGuid)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String, sqlOrderBy As String, callersCacheNameList As List(Of String)) As List(Of ContentFieldHelpModel)
            Return createList(Of ContentFieldHelpModel)(cpCore, sqlCriteria, sqlOrderBy, callersCacheNameList)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String, sqlOrderBy As String) As List(Of ContentFieldHelpModel)
            Return createList(Of ContentFieldHelpModel)(cpCore, sqlCriteria, sqlOrderBy)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createList(cpCore As coreClass, sqlCriteria As String) As List(Of ContentFieldHelpModel)
            Return createList(Of ContentFieldHelpModel)(cpCore, sqlCriteria)
        End Function
        '
        '====================================================================================================
        Public Overloads Sub invalidatePrimaryCache(cpCore As coreClass, recordId As Integer)
            invalidateCacheSingleRecord(Of ContentFieldHelpModel)(cpCore, recordId)
        End Sub
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, recordId As Integer) As String
            Return baseModel.getRecordName(Of ContentFieldHelpModel)(cpcore, recordId)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordName(cpcore As coreClass, ccGuid As String) As String
            Return baseModel.getRecordName(Of ContentFieldHelpModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function getRecordId(cpcore As coreClass, ccGuid As String) As Integer
            Return baseModel.getRecordId(Of ContentFieldHelpModel)(cpcore, ccGuid)
        End Function
        '
        '====================================================================================================
        Public Overloads Shared Function createDefault(cpcore As coreClass) As ContentFieldHelpModel
            Return createDefault(Of ContentFieldHelpModel)(cpcore)
        End Function
        '
        '====================================================================================================
        ''' <summary>
        ''' get the first field help for a field, digard the rest
        ''' </summary>
        ''' <param name="cpCore"></param>
        ''' <param name="fieldId"></param>
        ''' <returns></returns>
        Public Shared Function createByFieldId(cpCore As coreClass, fieldId As Integer) As ContentFieldHelpModel
            Dim helpList As List(Of ContentFieldHelpModel) = createList(cpCore, "(fieldId=" & fieldId & ")", "id")
            If (helpList.Count = 0) Then
                Return Nothing
            Else
                Return helpList.First
            End If
        End Function
    End Class
End Namespace

File: source/clibCoreCs/Models/Entity/contentFieldHelpModel.cs

using System;
using System.Reflection;
using System.Xml;
using System.Diagnostics;
using System.Linq;
using System.Text.RegularExpressions;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using Contensive.Core;
using Contensive.Core.Models.Entity;
using Contensive.Core.Controllers;
using static Contensive.Core.Controllers.genericController;
using static Contensive.Core.constants;
//
namespace Contensive.Core.Models.Entity {
    public class ContentFieldHelpModel : baseModel {
        //
        //====================================================================================================
        //-- const
        public const string contentName = "content field help";
        public const string contentTableName = "ccFieldHelp";
        private const string contentDataSource = "default";
        //
        //====================================================================================================
        // -- instance properties
        public int FieldID { get; set; }
        public string HelpCustom { get; set; }
        public string HelpDefault { get; set; }
        //
        //====================================================================================================
        public static ContentFieldHelpModel add(coreClass cpCore) {
            return add<ContentFieldHelpModel>(cpCore);
        }
        //
        //====================================================================================================
        public static ContentFieldHelpModel add(coreClass cpCore, ref List<string> callersCacheNameList) {
            return add<ContentFieldHelpModel>(cpCore, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public static ContentFieldHelpModel create(coreClass cpCore, int recordId) {
            return create<ContentFieldHelpModel>(cpCore, recordId);
        }
        //
        //====================================================================================================
        public static ContentFieldHelpModel create(coreClass cpCore, int recordId, ref List<string> callersCacheNameList) {
            return create<ContentFieldHelpModel>(cpCore, recordId, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public static ContentFieldHelpModel create(coreClass cpCore, string recordGuid) {
            return create<ContentFieldHelpModel>(cpCore, recordGuid);
        }
        //
        //====================================================================================================
        public static ContentFieldHelpModel create(coreClass cpCore, string recordGuid, ref List<string> callersCacheNameList) {
            return create<ContentFieldHelpModel>(cpCore, recordGuid, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public static ContentFieldHelpModel createByName(coreClass cpCore, string recordName) {
            return createByName<ContentFieldHelpModel>(cpCore, recordName);
        }
        //
        //====================================================================================================
        public static ContentFieldHelpModel createByName(coreClass cpCore, string recordName, ref List<string> callersCacheNameList) {
            return createByName<ContentFieldHelpModel>(cpCore, recordName, ref callersCacheNameList);
        }
        //
        //====================================================================================================
        public new void save(coreClass cpCore) {
            base.save(cpCore);
        }
        //
        //====================================================================================================
        public static void delete(coreClass cpCore, int recordId) {
            delete<ContentFieldHelpModel>(cpCore, recordId);
        }
        //
        //====================================================================================================
        public static void delete(coreClass cpCore, string ccGuid) {
            delete<ContentFieldHelpModel>(cpCore, ccGuid);
        }
        //
        //====================================================================================================
        public static List<ContentFieldHelpModel> createList(coreClass cpCore, string sqlCriteria, string sqlOrderBy, List<string> callersCacheNameList) {
            return createList<ContentFieldHelpModel>(cpCore, sqlCriteria, sqlOrderBy, callersCacheNameList);
        }
        //
        //====================================================================================================
        public static List<ContentFieldHelpModel> createList(coreClass cpCore, string sqlCriteria, string sqlOrderBy) {
            return createList<ContentFieldHelpModel>(cpCore, sqlCriteria, sqlOrderBy);
        }
        //
        //====================================================================================================
        public static List<ContentFieldHelpModel> createList(coreClass cpCore, string sqlCriteria) {
            return createList<ContentFieldHelpModel>(cpCore, sqlCriteria);
        }
        //
        //====================================================================================================
        public void invalidatePrimaryCache(coreClass cpCore, int recordId) {
            invalidateCacheSingleRecord<ContentFieldHelpModel>(cpCore, recordId);
        }
        //
        //====================================================================================================
        public static string getRecordName(coreClass cpcore, int recordId) {
            return baseModel.getRecordName<ContentFieldHelpModel>(cpcore, recordId);
        }
        //
        //====================================================================================================
        public static string getRecordName(coreClass cpcore, string ccGuid) {
            return baseModel.getRecordName<ContentFieldHelpModel>(cpcore, ccGuid);
        }
        //
        //====================================================================================================
        public static int getRecordId(coreClass cpcore, string ccGuid) {
            return baseModel.getRecordId<ContentFieldHelpModel>(cpcore, ccGuid);
        }
        //
        //====================================================================================================
        public static ContentFieldHelpModel createDefault(coreClass cpcore) {
            return createDefault<ContentFieldHelpModel>(cpcore);
        }
        //
        //====================================================================================================
        /// <summary>
        /// get the first field help for a field, digard the rest
        /// </summary>
        /// <param name="cpCore"></param>
        /// <param name="fieldId"></param>
        /// <returns></returns>
        public static ContentFieldHelpModel createByFieldId(coreClass cpCore, int fieldId) {
            List<ContentFieldHelpModel> helpList = createList(cpCore, "(fieldId=" + fieldId + ")", "id");
            if (helpList.Count == 0) {
                return null;
            } else {
                return helpList.First();
            }
        }
    }
}
